### PR TITLE
Add a basic Url rewrite modifier.

### PR DESCRIPTION
**Pull request by favalos**
Wrote a very simple and basic Url Rewrite functionality.

With the test configuration I tested doing:

http://localhost:8080/rewrite/blog/

Hope this could be the beginning of a contribution to this project, any feedback or improvements please let me know.

Thanks!

### CODE REVIEW

**Comment by jamesmunns** [comment]
Hey there! I think this looks reasonable, from a quick first glance:

* It would be good to remove any `unwrap`s, at least during the proxying stage (it's more okay at configuration/setup time as those will fire at startup). This isn't a hard blocker, but it'll help me out later when I go to cleanup the other existing unwrap/expects!
* It would be good to add this to the user manual, e.g. https://github.com/memorysafety/river/blob/main/user-manual/src/config/kdl.md#servicesnamepath-control

One perf note is that it might be faster to do the replace, then figure out if we replaced anything, instead of looking for the match and then replacing. [The docs](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace) say:

> If no match is found, then the haystack is returned unchanged. In that case, this implementation will likely return a Cow::Borrowed value such that no allocation is performed.
> 
> When a Cow::Borrowed is returned, the value returned is guaranteed to be equivalent to the haystack given.

That being said: I'm just guessing (and that's not a good basis for requesting a perf change!) that it'd be faster to do a replace and then check if the result is Borrowed or Owned, instead of doing a match and then conditionally a replace, so maybe just a comment that we could consider benchmarking this in the future would be fine.

Basically suggesting:

```rust
// instead of this:
if is_match() {
    replace()
}

// this
let res = replace()
if !res.is_borrowed() {
    // set the new URI here
}
```

**Review by jamesmunns — CHANGES_REQUESTED** [review]
Made some nits, the one blocker is that I'd like to see the docs updated, then we can merge this!

**Comment by Szpadel** [comment]
My 2 cents about performance
Instead of 
```
filter kind="url-rewrite" regex="^/rewrite/(.*)$" rewrite="/${1}"
```

Doing
```
filter kind="url-rewrite" regex="^/rewrite/" rewrite="/"
```
So removing capture group would switch `replacen` , with `replace` is wrapping, to fast path with significally better performance.
(I think that this could be documented)

```rust
if !res.is_borrowed() {
    // set the new URI here
}
```
I feel here more readable would be `res.is_owned()` instead. Or `match` if we want to avoid nightly only feature.
